src/client/pool.rs

```diff
@@ -721,7 +723,8 @@ impl Expiration {
 
     fn expires(&self, instant: Instant) -> bool {
         match self.0 {
-            Some(timeout) => instant.elapsed() > timeout,
+            // Avoid `Instant::elapsed` to avoid issues like rust-lang/rust#86470.
+            Some(timeout) => Instant::now().saturating_duration_since(instant) > timeout,
```

Comment:
It definitely feels like `Instant::elapsed()` should handle this, or otherwise it's not really safe to use. Looks like https://github.com/rust-lang/rust/pull/89926 agrees.

Reply:
Yeah, I agree. It will probably take at least a few months for those changes to be available in a stable release (the RFC still appears to be open). This change would eliminate these problems well in advance of that, even for projects that are not on the latest stable Rust version.

Is there something I can do to make the temporary nature of this hack/fix better documented?